### ISSUE TITLE
Automation shell override causes invalid options.shell property

### ISSUE THREAD

**Issue by edemaine** [microsoft/vscode-makefile-tools#346]
My `settings.json` includes the following:

```json
    "terminal.integrated.automationProfile.windows": {
      "path": "${env:windir}\\System32\\cmd.exe",
      "args": [],
      "icon": "terminal-cmd"
    },
```

I admit I'm pretty confused about whether this option is supposed to be a string (a key of `terminal.integrated.profiles.windows`) or an object, but the TypeScript types suggest it's supposed to be an object.  In this state, the Makefile extension says this:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "options.shell" property must be one of type boolean or string. Received an instance of Object
Preprocessing: "undefined"
Configure failed.
Configure elapsed time: 0.388
Dryrun timeout. Verify that the make command works properly in your development terminal (it could wait for stdin).
```

(No `dryrun.log` was generated.)

**Comment by andreeis** [comment]
@edemaine, at the time we implemented some functionality around this setting I wasn't able to find clear documentation about how it's defined. We fixed another suggestion from the community and in that case I saw an example where this was treated as a string. Also there was an impact on the execution (the issue was https://github.com/microsoft/vscode-makefile-tools/issues/233), we fixed something so this for sure can be a string... but we can look into recognizing the other alternative type.
I'll read this vscode issue closely: https://github.com/microsoft/vscode/issues/140261.

**Comment by wqferr** [comment]
I'm having the same issue, has this been fully patched?

**Comment by gcampbell-msft** [comment]
@wqferr No, this issue has not been fixed. There have been updates to the recommended settings for these options and we haven't yet updated based on the most recent documentation. Additionally, there are decisions to be made about what settings we will honor and which we will not. 

It is on our backlog and we will continue to track where it falls amongst our other priorities. Thanks!

**Comment by mouse07410** [comment]
Could you please prioritize this issue? Thanks!

**Comment by ivadmitry** [comment]
Adding these lines to settings.json works as W/A to me:
        "terminal.integrated.automationProfile.linux": null,
        "terminal.integrated.automationShell.linux": "true",

Probably the issue is that shell argument of child_process.spawn() should be a Boolean, in src/util.ts

**Comment by Yingzi1234** [comment]
@gcampbell-msft We can still reproduce this issue in the latest version, you can get the details below.

**Preparation:**
Makefile Toos: v0.10.5(pre-release)
C/C++: v1.20.1
VSCode: v1.88.1

**Repro steps:**
1. Clone a repro and open it
2. Open the VS Code settings and set `makefile.extensionOutputFolder` to “.vscode”.
3. Add the following command to _settings.json_ file
     "terminal.integrated.automationProfile.windows": {
      "path": "${env:windir}\\System32\\cmd.exe",
      "args": [],
      "icon": "terminal-cmd"
    },
4. Clcik F1, run the "Makefile: Configure" command

**Actual result:**
Configuration will fail
![image](https://github.com/microsoft/vscode-makefile-tools/assets/101234795/acbecc80-e6ab-4020-a9d9-988bcac6ff5e)
